timesketch/api/v1/resources/sigma.py

```diff
@@ -87,15 +125,274 @@ def get(self, rule_uuid):
                     return_rule = rule
 
         if return_rule is None:
-            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sigma rule found with this ID.")
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND, "No sigma rule found with this ID."
+            )
 
-        meta = {"current_user": current_user.username, "rules_count": len(sigma_rules)}
+        meta = {
+            "current_user": current_user.username,
+            "rules_count": len(sigma_rules),
+        }
         return jsonify({"objects": [return_rule], "meta": meta})
 
 
+# TODO(jaegeral): deprecate this class
 class SigmaByTextResource(resources.ResourceMixin, Resource):
     """Resource to get a Sigma rule by text."""
 
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+        Returns:
+            JSON sigma rule
+        """
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        content = form.get("content")
+        if not content:
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Missing values from the request.",
+            )
+
+        try:
+            sigma_rule = ts_sigma_lib.parse_sigma_rule_by_text(content)
+
+        except ValueError:
+            logger.error(
+                "Sigma Parsing error with the user provided rule",
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Error unable to parse the provided Sigma rule",
+            )
+
+        except NotImplementedError as exception:
+            logger.error(
+                "Sigma Parsing error: Feature in the rule provided "
+                " is not implemented in this backend",
```

Comment:
Storing these error messages in an intermediary variable would reduce deduplication and inconsistencies in error messages (e.g. L166 / L171)